### PR TITLE
refactor: adjust slack messages to add link to job

### PR DESCRIPTION
**Pull request by nathan-weinberg**
**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: .github/workflows/e2e-nvidia-a10g-x4.yml [hunk 1]
```diff
@@ -164,16 +164,29 @@ jobs:
         env:
           GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
-      - name: Post job results to Slack
-        if: ${{ !cancelled() && steps.check_pr.outputs.is_pr == 'false' }}
-        id: slack
+      - name: Post job results to Slack if the workflow failed
+        if: failure() && steps.check_pr.outputs.is_pr == 'false'
+        id: slack-failure
```
Comment (danmcp) [inline .github/workflows/e2e-nvidia-a10g-x4.yml:169]: Would slack-report-failure be more appropriate?

Comment (nathan-weinberg) [reply .github/workflows/e2e-nvidia-a10g-x4.yml]: SGTM

**Review by danmcp — APPROVED** [review]

**Review by leseb — DISMISSED** [review]

File: .github/workflows/e2e-nvidia-a10g-x4.yml [hunk 2]
```diff
@@ -164,16 +164,29 @@ jobs:
...
+        if: failure() && steps.check_pr.outputs.is_pr == 'false'
+        id: slack-report-failure
         uses: slackapi/slack-github-action@37ebaef184d7626c5f204ab8d3baff4262dd30f0 # v1.27.0
         with:
           # Slack channel id, channel name, or user id to post message.
           # See also: https://api.slack.com/methods/chat.postMessage#channels
           # You can pass in multiple channels to post to by providing a comma-delimited list of channel IDs.
           channel-id: 'cli-e2e-results'
-          slack-message: "*e2e-nvidia-a10g-x4* job in *${{ github.repository }}* running on *${{ format('Branch {0}', steps.check_pr.outputs.pr_or_branch) }}* completed with result: *${{ job.status }}*"
+          slack-message: "*e2e-nvidia-a10g-x4* job in *${{ github.repository }}* running on branch `${{ steps.check_pr.outputs.pr_or_branch }}` completed *with failures* :meow_sad-rain: | ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}"
```
[the remark this answers is not in the document]
Comment (nathan-weinberg) [reply .github/workflows/e2e-nvidia-a10g-x4.yml]: Pushed as a second commit, can squash later

[the remark this answers is not in the document]
Comment (nathan-weinberg) [reply .github/workflows/e2e-nvidia-a10g-x4.yml]: @leseb not sure if this suggestion will work: https://github.com/instructlab/instructlab/actions/runs/10774544678/job/29876856552?pr=2215

[the remark this answers is not in the document]
Comment (leseb) [reply .github/workflows/e2e-nvidia-a10g-x4.yml]: :/ nevermind then.

**Review by danmcp — APPROVED** [review]

**Pull request merged**